### ISSUE TITLE
Reconstruction loss compared to metaquery

### ISSUE THREAD

**Issue by toyot-li** [HorizonWind2004/reconstruction-alignment#21]
Hi @HorizonWind2004 Thanks for your great work!

I find that MetaQuery also uses reconstruction loss but shows moderate improvement (refer to appendix C of metaquery paper)

May I know your opinion?

**Comment by HorizonWind2004** [comment]
Hello! Thank you for your interest!

MetaQueries put reconstruction into pretraining stage, and the conclusion is "no improvement".

However, OpenUni is a open-source version of MetaQueries that do not trained on reconstruction task in the pretraining stage. And we surprisingly found that, if you add it to the post-training stage, it shows a huge improvement on different image generation benchmark (GenEval, WISE, DPG, T2ICompBench).

So it is a very interesting topic that worth exploring that: (1) Why post-training works? (2) How to make pre-training works?

My personal opinion is that, during the pretraining stage, model overfits to reconstruction, and learns no knowledge from visual understanding but degrades to an auto-encoder.
